RecoTracker/LST/plugins/LSTOutputConverter.cc

```diff
@@ -0,0 +1,277 @@
+#include "DataFormats/TrackerRecHit2D/interface/Phase2TrackerRecHit1D.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHitCollection.h"
+#include "DataFormats/TrackCandidate/interface/TrackCandidateCollection.h"
+#include "DataFormats/TrackReco/interface/SeedStopInfo.h"
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "Geometry/CommonDetUnit/interface/GeomDet.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+#include "RecoTracker/LST/interface/LSTPhase2OTHitsInput.h"
+#include "RecoTracker/LST/interface/LSTOutput.h"
+#include "RecoTracker/TkSeedingLayers/interface/SeedingHitSet.h"
+
+#include "RecoTracker/TkSeedGenerator/interface/SeedCreator.h"
+#include "RecoTracker/TkSeedGenerator/interface/SeedCreatorFactory.h"
+
+#include "RecoTracker/TkTrackingRegions/interface/GlobalTrackingRegion.h"
+#include "TrackingTools/GeomPropagators/interface/Propagator.h"
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+
+class LSTOutputConverter : public edm::global::EDProducer<> {
+public:
+  explicit LSTOutputConverter(edm::ParameterSet const& iConfig);
+  ~LSTOutputConverter() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const override;
+
+  const edm::EDGetTokenT<LSTOutput> lstOutputToken_;
+  const edm::EDGetTokenT<LSTPhase2OTHitsInput> lstPhase2OTHitsInputToken_;
+  const edm::EDGetTokenT<TrajectorySeedCollection> lstPixelSeedToken_;
+  const bool includeT5s_;
+  const bool includeNonpLSTSs_;
+  const edm::ESGetToken<MagneticField, IdealMagneticFieldRecord> mfToken_;
+  edm::ESGetToken<Propagator, TrackingComponentsRecord> propagatorAlongToken_;
+  edm::ESGetToken<Propagator, TrackingComponentsRecord> propagatorOppositeToken_;
+  const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> tGeomToken_;
+  std::unique_ptr<SeedCreator> seedCreator_;
```

Comment:
there was a response to this comment on why this was not addressed in Jun 14 update
https://github.com/cms-sw/cmssw/pull/45117#issuecomment-2169002630
Sorry, it got misplaced in a batch response.

I'm copy-pasting it below.

This is not addressed because line:
https://github.com/SegmentLinking/cmssw/blob/314e95d4162a0ece6a047a6d04d1731221c66e9a/RecoTracker/LST/plugins/LSTOutputConverter.cc#L202
requires non-constness for SeedCreator, as far as we understand (and the code errors tell so).

Reply:
Thanks, I missed the earlier response. If a non-`const` access to `SeedCreator` is needed, then this object is potentially thread-unsafe (and looking at the concrete `SeedFromConsecutiveHitsCreator` and `CosmicSeedCreator` confirms it indeed is thread-unsafe), and must to be moved to [StreamCache](https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkGlobalModuleInterface#edm_StreamCacheT) (or the entire module is changed to an `edm::stream` type; `edm::global` + `StreamCache` would be preferred although in this case the difference in memory usage seems likely small.)